index.js

```diff
@@ -10,9 +10,9 @@
  */
 
 if (!Array.prototype.forEachAsync) {
-    Array.prototype.forEachAsync = async function(callback) {
-        for (let index = 0; index < this.length; index++) {
-            await callback(this[index], index, this);
-        }
+    Array.prototype.forEachAsync = async function (callback) {
+        await Promise.all(this.map((item, index) => {
+            return callback(item, index, this);
+        }));
```

Comment:
This modification shifts the behavior from sequential execution to parallel execution, potentially leading to unforeseen outcomes. It is recommended to introduce a new parameter to accommodate and facilitate this functionality. e.g. 
```
Array.prototype.forEachAsync = async function (callback, parallel = false) {
  if (parallel) {
      // new logical
  } else {
     // old logical
  }
}
```
comments should also be updated.

Reply:
> This modification shifts the behavior from sequential execution to parallel execution, potentially leading to unforeseen outcomes. It is recommended to introduce a new parameter to accommodate and facilitate this functionality. e.g.
> 
> ```
> Array.prototype.forEachAsync = async function (callback, parallel = false) {
>   if (parallel) {
>       // new logical
>   } else {
>      // old logical
>   }
> }
> ```
> 
> comments should also be updated.

Done :)